### ISSUE TITLE
New Feature: Support for git flow/ git flow action buttons

### ISSUE THREAD

**Issue by teebee76** [jzoss/Git-Source-Control-Provider#31]
Besides the gscp plugin I have been using sourcetree as a tool for managing my git repositories. In Sourcetree they've implemented a button for git flow actions. What it actually does, is a series of common git -operations put in sequence, and it save either a lot of typing in git bash, or a lot of clicking inside pending changes.

In short it is a system to easily manage the use of a branching model for deployment.
You can read about the model here: http://nvie.com/posts/a-successful-git-branching-model/ and it also shows a nice big chart you can put on your wall :)

So how this makes things easier... When I click my Git Flow button (at a state where I have no release- or feature- or hotfix- branches, I am given the option to start a new release/feature/hotfix. If it is a hotfix, it will branch off my master branch else it will branch off my develop branch. The name I give will be appended to release-/feature-/hotfix-. Nothing really magical about this step, I guess it's just a new branch (although it will first checkout the correct branch, If I'm not already on it) 

I then do the coding required, with as many normal commits etc as needed.

When I am finished I click git flow again, and it will recognice that I have a branch called hotfix- for instance, and will present the option of Finishing off this branch. If chosen I can give a tag etc to mark the commit and it will merge my branch into the master (for hotfix and release) and also to the develop branch. After that it will remove the hotfix branch.

I also have options etc for pushing to remote.

So is this something people will find useful?

Should it be in the gscp or should it rather be a plugin like git-extensions that will get it's own menu in pending changes?

**Comment by teebee76** [comment]
After a little searching I found this extension **[GitFlow for Visual Studio 2015]([https://visualstudiogallery.msdn.microsoft.com/f5ae0a1d-005f-4a09-a19c-3f46ff30400a])** 
It seems to work along side gscp although I have to open the TeamExplorer window to use it.
I don't know if it could be used as a plugin to gscp to get a context menu like for git Extensions?
I will test it for a little while, and see if it is more working well or just a hazzle to use it. 

**...tbc...**

**Comment by jzoss** [comment]
@teebee76 Yeah I saw that,, Thought it would not be a nice integration just to say have them side by side.. But I have a sub-project I may start.. I think you may be the person to help me.. So I think if we were to make this part of this.. the first thing to do would be to start project to basically wrap git flow around libgit2sharp.. There are not that many commands.. But that would give us a native way of doing it.. Then we just would have to add the UI to the Scc plugin .. Want to help :)

**Comment by teebee76** [comment]
Sounds like a good idea, I will do what I can, hopefully I can structure my time better to get more time to do fun stuff like this. 
Never done any extension-stuff in VS before though, so I might need some pointers in the beginning to get things setup correctly. But I will investigate a little tomorrow if I get the tiime :)

**Comment by jzoss** [comment]
First thing to do is not do the work in the vs extension :) First.. I started a new [project](https://github.com/jzoss/libgitflow2sharp).. Which I have not done anything to yet.. I can make a contributor if you want  .. Anyway I want to do extension methods I think for libgit2sharp to to all the gitflow commands... or so I'm thinking right now.

**Comment by teebee76** [comment]
![image](https://cloud.githubusercontent.com/assets/5390116/13048652/9628218c-d3e7-11e5-91fc-ae62075bdcf0.png)
;)

**Comment by yannduran** [comment]
@jzoss +1 for integrating GitFlow into your extension. I'd be happy to give whatever assistance that I can, if you'll have me that is :-)

**Comment by jzoss** [comment]
@yannduran This is totally the next feature I want to implement. So.. First things first.. I was working on making GitFlow work natively with libgit2sharp .. I actually made it pretty far.. Check out this repo... https://github.com/jzoss/libgitflow2sharp .. Once that is done.. It I want to start adding it to the extension.
